safenode/src/bin/kadnode.rs

```diff
@@ -28,12 +27,25 @@ async fn main() -> Result<()> {
     let _log_appender_guard = init_node_logging(&opt.log_dir)?;
 
     let socket_addr = SocketAddr::new(opt.ip, opt.port);
+    let peers = parse_peer_multiaddreses(&opt.peers)?;
 
     info!("Starting a node...");
-    let node_events_channel = Node::run(socket_addr).await?;
+    let node_events_channel = Node::run(socket_addr, peers).await?;
 
     let mut node_events_rx = node_events_channel.subscribe();
-    if let Ok(event) = node_events_rx.recv().await {
+
+    loop {
+        let event = match node_events_rx.recv().await {
+            Ok(e) => e,
+            Err(tokio::sync::broadcast::error::RecvError::Closed) => {
+                tracing::error!("Node event channel closed!");
+                break;
+            }
+            Err(tokio::sync::broadcast::error::RecvError::Lagged(n)) => {
+                tracing::warn!("Skipped {n} node events!");
+                continue;
+            }
+        };
         match event {
             NodeEvent::ConnectedToNetwork => {
                 info!("Connected to the Network");
```

Comment:
Looks like we only ever break from this infinite loop when `Err(tokio::sync::broadcast::error::RecvError::Closed)`, only to block forever in a sleep loop below.
On the other hand, when we have `NodeEvent::ConnectedToNetwork` we don't break out.
I have trouble understanding what this code is doing. Am I missing something?